src/content/app/tools/blast/views/blast-form/BlastForm.tsx

```diff
@@ -60,12 +60,23 @@ const BlastForm = () => {
 
 const Main = () => {
   const isSmallViewport = useMediaQuery('(max-width: 1900px)');
+  const [sequenceValidityStatus, setSequenceValidityStatus] = useState({});
+
+  const updateSequenceValidity = (index: number, status: boolean) => {
+    setSequenceValidityStatus({ ...sequenceValidityStatus, [index]: status });
```

Comment:
so what do we do with keys for indexes that no longer exist?

Reply:
I am still not seeing in the code anything that would change the size of the array based on the length of the sequences plus the presence/absence of the empty input box.